google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineArtifactDeploymentSourceType.java

```diff
@@ -93,12 +85,16 @@ public void save(@NotNull ArtifactDeploymentSource deploymentSource,
       AppEngineDeployable deployable = (AppEngineDeployable) deploymentSource;
 
       if (deployable.getProjectName() != null) {
-        tag.setAttribute(PROJECT_ATTRIBUTE,
-            ((AppEngineDeployable) deploymentSource).getProjectName());
+        tag.setAttribute(PROJECT_ATTRIBUTE, deployable.getProjectName());
       }
 
       if (deployable.getVersion() != null) {
-        tag.setAttribute(VERSION_ATTRIBUTE, ((AppEngineDeployable) deploymentSource).getVersion());
+        tag.setAttribute(VERSION_ATTRIBUTE, deployable.getVersion());
+      }
+
+      if (deployable.getEnvironment() != null) {
```

Comment:
For Standard apps, it is. But we need this information for Flexible apps, too, and we also need it for flow control in a few other places. Off the top of my head, Flex Maven deployments were failing because the environment wasn't being serialized.